Source/Core/Editor/TextToSpeech/TextToSpeechBuildPreprocessor.cs

```diff
@@ -9,14 +18,37 @@ namespace VRBuilder.Core.Editor.TextToSpeech
     /// </summary>
     public class TextToSpeechBuildPreprocessor : IPreprocessBuildWithReport
     {
-        public int callbackOrder => 1;
+        public int callbackOrder => 0;
 
-        /// <summary>
-        /// Generates TTS files for all processes before a build.
-        /// </summary>
         public void OnPreprocessBuild(BuildReport report)
         {
-            TextToSpeechEditorUtils.GenerateTextToSpeechForAllProcesses();
+            if (TextToSpeechSettings.Instance.GenerateAudioInBuildingProcess)
+            {
+                Task task = TextToSpeechEditorUtils.GenerateTextToSpeechForAllProcesses();
+                task.Wait();
+                
+                //TODO enable TTS building for active scenes only
+                //foreach (var sceneItem in EditorBuildSettings.scenes)
+                //{
+                //    Scene scene = SceneManager.GetSceneByPath(sceneItem.path);
+                //    var runtimeConfigurator = scene.GetRootGameObjects()
+                //        .Select(go => go.GetComponentInChildren<RuntimeConfigurator>(true))
+                //        .FirstOrDefault(c => c != null);
+                //    if (runtimeConfigurator != null)
+                //    {
+                //        Task task = TextToSpeechEditorUtils.GenerateTextToSpeechForProcess(runtimeConfigurator.GetSelectedProcess());
+                //        task.Wait();
+                //    }
+                //    else
+                //    {
+                //        UnityEngine.Debug.LogWarning($"No TTS files for the scene {scene.name} could be generated. Error RuntimeConfigurator component was not found in the scene.");
+                //    }
+                //}
+            }
+            else
+            {
+                UnityEngine.Debug.LogWarning("Text to speech files generation is disabled.");
```

Comment:
```suggestion
                UnityEngine.Debug.LogWarning("Automated text to speech files generation is disabled. Please ensure files are up to date before building!");
```
I'd make this message a little clearer.